### PR TITLE
[rstmgr] Streamline rstmgr

### PR DESCRIPTION
**Pull request by tjaychen**
Address #3089 

This combines the reset regen and software reset controls all into one register.
Since multireg doesn't really support this function, it is handled outside reg for now.

### CODE REVIEW

**Comment by tjaychen** [comment]
@lenary @silvestrst 
can you guys have a look and the updated enums? hope this is clear.

File: util/topgen/c.py [hunk 1]
```diff
@@ -368,3 +369,15 @@ def _init_pwrmgr_wakeups(self):
...
+    # Enumerates the positions of all software controllable resets
+    def _init_rstmgr_sw_rsts(self):
+        sw_rsts = [rst for rst in self.top["resets"]["nodes"] if 'sw' in rst
+                   and rst['sw'] == 1]
+
+        enum = CEnum(self._top_name + Name(["reset", "manager", "sw", "resets"]))
+
+        for rst in sw_rsts:
+            enum.add_constant(Name.from_snake_case(rst["name"]))
+
```
Comment (lenary) [inline util/topgen/c.py:382]: please do `enum.add_last_constant(<docstring>)` but it's looking good.

**Comment by tjaychen** [comment]
o that's right. Thanks for the reminder.

**Review by eunchan — APPROVED** [review]
RTL and template part looks good. Do you also have a plan to revise the reset contraints in SDC  based on the input or manual editing as same as bronze?

**Comment by tjaychen** [comment]
> RTL and template part looks good. Do you also have a plan to revise the reset contraints in SDC based on the input or manual editing as same as bronze?

ahh...i completely forgot about that....do we have reset constraints for like..ideal nets? Because i didn't change the final reset mux, so those should technically still apply i think.

**Pull request merged**